packages/docusaurus-theme-classic/src/nprogress.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import nprogress from 'nprogress';
+import './nprogress.css';
+import type {Location} from 'history';
+
+nprogress.configure({showSpinner: false});
+
+const delay = 200;
+
+export function onRouteUpdate({
+  location,
+  previousLocation,
+}: {
+  location: Location;
+  previousLocation: Location | null;
+}): (() => void) | undefined {
+  if (location.pathname !== previousLocation?.pathname) {
+    const progressBarTimeout = window.setTimeout(() => {
+      nprogress.start();
+    }, delay);
+    return () => window.clearTimeout(progressBarTimeout);
```

Comment:
Technically, we can also stop the progress bar here, because at this time, the page is just about to re-render. But we can also use this to test `onRouteDidUpdate`.

Reply:
Let's keep it this way. Doesn't look too problematic